fakturogen-invoice/src/test/java/pl/fakturogen/invoice/service/mapper/InvoiceTemplateMapperTest.java

```diff
@@ -0,0 +1,150 @@
+import pl.fakturogen.invoice.web.dto.CustomerDTO;
+import pl.fakturogen.invoice.web.dto.InvoiceTemplateDTO;
+import pl.fakturogen.invoice.web.dto.ProductDTO;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class InvoiceTemplateMapperTest {
+
+    private InvoiceTemplate invoiceTemplate;
+    private InvoiceTemplate invTempEmpty;
+    private InvoiceTemplateDTO invoiceTemplateDTO;
+    private InvoiceTemplateDTO invTempEmptyDTO;
+    private InvoiceTemplateMapper invoiceTemplateMapper;
+
+    @BeforeEach
+    public void init() {
+        invoiceTemplateMapper = new InvoiceTemplateMapper();
+
+        LocalDate issueDate = LocalDate.of(2021, 1, 20);
+        LocalDate saleDate = LocalDate.of(2021, 2, 1);
+        Customer customer = new Customer();
+        customer.setName("CUSTOMER NAME");
+        CustomerDTO customerDTO = new CustomerDTO();
+        customerDTO.setName("CUSTOMER NAME");
+        Product product1 = new Product();
+        product1.setName("PRODUCT ONE");
+        Product product2 = new Product();
+        product2.setName("TWO PRODUCT");
+
+        ProductDTO productDTO1 = new ProductDTO();
+        productDTO1.setName("PRODUCT ONE");
+        ProductDTO productDTO2 = new ProductDTO();
+        productDTO2.setName("TWO PRODUCT");
+
+        List<Product> productList = new ArrayList<>();
+        productList.add(product1);
+        productList.add(product2);
+
+        List<ProductDTO> productDTOList = new ArrayList<>();
+        productDTOList.add(productDTO1);
+        productDTOList.add(productDTO2);
+
+        invoiceTemplate = new InvoiceTemplate();
+        invoiceTemplate.setId(1L);
+        invoiceTemplate.setIssueDate(issueDate);
+        invoiceTemplate.setSaleDate(saleDate);
+
+        invoiceTemplate.setPaymentMethod("TEST PAYMENT METHOD");
+        invoiceTemplate.setPriceTotal(999_111.0);
+        invoiceTemplate.setPriceTax(111_111.0);
+        invoiceTemplate.setPriceNet(888_000.0);
+        invoiceTemplate.setDiscountAmount(20.00);
+        invoiceTemplate.setCustomer(customer);
+        invoiceTemplate.setProducts(productList);
+
+        invoiceTemplateDTO = new InvoiceTemplateDTO();
+        invoiceTemplateDTO.setId(1L);
+        invoiceTemplateDTO.setIssueDate(issueDate);
+        invoiceTemplateDTO.setSaleDate(saleDate);
+        invoiceTemplateDTO.setPaymentMethod("TEST PAYMENT METHOD");
+        invoiceTemplateDTO.setPriceTotal(999_111.0);
+        invoiceTemplateDTO.setPriceTax(111_111.0);
+        invoiceTemplateDTO.setPriceNet(888_000.0);
+        invoiceTemplateDTO.setDiscountAmount(20.00);
+        invoiceTemplateDTO.setCustomer(customerDTO);
+        invoiceTemplateDTO.setProducts(productDTOList);
+
+        invTempEmpty = new InvoiceTemplate();
+        invTempEmptyDTO = new InvoiceTemplateDTO();
+    }
+
+    @Nested
+    class invoiceTemplateToDTO {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).